lib/domain/services/execution_services.dart

```diff
@@ -0,0 +1,163 @@
+import 'package:memo/data/repositories/collection_repository.dart';
+import 'package:memo/data/repositories/memo_execution_repository.dart';
+import 'package:memo/data/repositories/memo_repository.dart';
+import 'package:memo/data/repositories/user_repository.dart';
+import 'package:memo/domain/enums/memo_difficulty.dart';
+import 'package:memo/domain/isolated_services/memory_stability_services.dart';
+import 'package:memo/domain/models/collection.dart';
+import 'package:memo/domain/models/memo.dart';
+import 'package:memo/domain/models/memo_execution.dart';
+
+/// Handles all domain-specific operations pertaining to the act of executing a [Collection] and its [Memo]s
+abstract class ExecutionServices {
+  /// Retrieves a suitable chunk of [Memo]s when executing a [Collection] of [collectionId]
```

Comment:
Is this function used DURING an execution? If it is, disregard the following suggestion.

IMO this could mention that the this chunk can be used - in the future - in an execution flow instead using `when executing`. Using `when executing` suggests to me that this chunk can be used during an execution instead before it.

I suggest something like:
`Retrieves a suitable chunk of [Memo]s of [collectionId] to be executed`

Reply:
Not during, just before executing.
The execution doesn't change anything when it's executing, that's a decision that I've made to make things way easier and should be implied when the user decides to leave an execution session, as a form of "discarding what he is doing at the moment", even though he may have executed one or more memos while at it.

Reply:
So considering that this function is always called right before an execution, and never during it, what do you think about my suggestion of updating the function docs to emphasis that this chunk of memos will be used in a execution that is to come instead using `when executing`?